### PR TITLE
Combine all script tests into `test-scripts`

### PR DESCRIPTION
**Pull request by mrobinson**
Remove:
    - tidy self test support from `./mach test`
    - `./mach test-idl`

Adds a `./mach test-scripts` command that is responsible for running all
Python script tests. Run this during the CI to catch regressions in
changes to scripts. The WebIDL tests are still *very* slow and there are
from Gecko, so only run them when "-a" is passed meaning all tests.

---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] There are tests for these changes

### CODE REVIEW

File: python/servo/testing_commands.py [hunk 1]
```diff
@@ -80,25 +78,12 @@ def __init__(self, context):
     @CommandArgument('params', default=None, nargs="...",
                      help="Optionally select test based on "
                           "test file directory")
-    @CommandArgument('--render-mode', '-rm', default=DEFAULT_RENDER_MODE,
```
Comment (mrobinson) [inline python/servo/testing_commands.py:83]: I removed some of these arguments as well that were not used.

**Review by jdm — APPROVED** [review]

**Comment by mrobinson** [comment]
This failed because I tried to call mach incorrectly on the Windows CI. Upon thinking about this a bit, I have removed the `test-script` call from the Window CI entirely. I'm not sure if all the tests pass on Windows and it's probably better not to risk it.

**Pull request merged**